eventlogger/src/main/java/gov/cms/ab2d/eventlogger/eventloggers/kinesis/KinesisEventLogger.java

```diff
@@ -1,63 +1,54 @@
-import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
 import gov.cms.ab2d.eventlogger.EventLogger;
 import gov.cms.ab2d.eventlogger.LoggableEvent;
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
 import org.springframework.stereotype.Service;
 
-import java.nio.ByteBuffer;
+import java.util.concurrent.Future;
 
 @Service
 @Slf4j
 public class KinesisEventLogger implements EventLogger {
     @Value("${execution.env:dev}")
     private String appEnv;
-
-    @Autowired
-    private AmazonKinesisFirehose client;
-
     @Value("${eventlogger.kinesis.stream.prefix:bfd-insights-ab2d-}")
     private String streamId;
 
+    private final KinesisConfig config;
+    private final AmazonKinesisFirehose client;
+
+    public KinesisEventLogger(KinesisConfig config, AmazonKinesisFirehose client) {
+        this.config = config;
+        this.client = client;
+    }
+
     @Override
     public void log(LoggableEvent event) {
-        String json = null;
+        log(event, false);
+    }
+
+    public void log(LoggableEvent event, boolean block) {
         event.setEnvironment(appEnv);
-        try {
-            json = getJsonString(event) + "\n";
-            Record record = new Record()
-                    .withData(ByteBuffer.wrap(json.getBytes()));
-
-            PutRecordRequest putRecordRequest = new PutRecordRequest();
-            String className = event.getClass().getSimpleName().toLowerCase();
-            putRecordRequest.setDeliveryStreamName(streamId + className);
-            putRecordRequest.setRecord(record);
-
-            PutRecordResult putRecordResult = client.putRecord(putRecordRequest);
-            ResponseMetadata data = putRecordResult.getSdkResponseMetadata();
-            event.setAwsId(data.getRequestId());
-        } catch (Exception ex) {
-            log.error("Error in pushing event to Kinesis " + json + " - " + ex.getMessage());
+        ThreadPoolTaskExecutor ex = config.kinesisLogProcessingPool();
+        KinesisEventProcessor processor = new KinesisEventProcessor(event, client, streamId);
+        Future<Void> future = ex.submit(processor);
+        if (block) {
+            while (!future.isDone()) {
+                try {
+                    Thread.sleep(10);
+                } catch (InterruptedException e) {
+                    log.error("Unable to wait for thread to sleep", e);
+                }
+            }
         }
     }
 
-    static String getJsonString(LoggableEvent event) throws JsonProcessingException {
-        ObjectMapper mapper = new ObjectMapper()
-                .registerModule(new ParameterNamesModule())
-                .registerModule(new Jdk8Module())
-                .registerModule(new JavaTimeModule());
-        return mapper.writeValueAsString(event);
+    boolean isFinished() {
```

Comment:
Is this used outside of testing? Perhaps put a @VisibleForTesting on it

Reply:
In theory, it could be. I had thought about adding it to the shutdown process.